src/fit.jl

```diff
@@ -91,16 +91,23 @@ function train!(
 
     start_iter = get_iter_number(estimator) + 1
     end_iter = start_iter + num_iterations - 1
+    total_metrics_evals = sum((((start_iter:end_iter) .- 1) .% estimator.metric_freq) .== 0)
 
     for iter in start_iter:end_iter
+
         is_finished = LGBM_BoosterUpdateOneIter(estimator.booster)
-        log_debug(verbosity, Dates.CompoundPeriod(now() - start_time),
-                  " elapsed, finished iteration ", iter, "\n")
+
+        log_debug(verbosity, Dates.CompoundPeriod(now() - start_time), " elapsed, finished iteration ", iter, "\n")
+
+        metric_idx = sum((((start_iter:iter) .- 1) .% estimator.metric_freq) .== 0)
```

Comment:
This one tells us the index to save into for store scores. We're only saving every metric_freq iterations and we're only saving for the batch of iterations we're currently doing